client_integration_test.go

```diff
@@ -651,6 +651,50 @@ func TestClientRemove(t *testing.T) {
 	}
 }
 
+func TestRemoveAll(t *testing.T) {
+	sftp, cmd := testClient(t, READWRITE, NODELAY)
+	defer cmd.Wait()
+	defer sftp.Close()
+
+	// Create a temporary directory for testing
+	tempDir, err := ioutil.TempDir("", "sftptest-removeAll")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(tempDir)
+
+	// Create file and directory within the temporary directory
+	f, err := ioutil.TempFile(tempDir, "sftptest-removeAll*.txt")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer f.Close()
+
+	d, err := ioutil.TempDir(tempDir, "sftptest-removeAll1")
+	if err != nil {
+		t.Fatal(err)
+	}
```

Comment:
This test would accidentally pass even if it did not have recursion, since as I noted in the code review comments, `Client.Remove()` would remove an empty directory.

Please build a directory tree that has at least a depth of 3. So `/tmp/sftptest-removeAll/foo/bar/baz`

Reply:
> Remove removes the specified file _or directory_. An error will be returned if no file or directory with the specified path exists, or if the specified directory is not empty. [emphasis added]

https://pkg.go.dev/github.com/pkg/sftp#Client.Remove

Reply:
Yes, I have red this documentation but it is not working as expected atleast in my case

Reply:
That’s really weird. Let’s file an `Issue` and investigate that. That should be the proper functioning.